proposals/devcontainer-distribution.md

```diff
@@ -0,0 +1,70 @@
+# Community contributions of templates and features
+
+This proposal is created to arrive at a schema where different communities and organizations can create features and dev container templates and make them discoverable to users.
+
+The proposal centers on the following ideas:
+
+- Discoverability for users both on the web (github.com) and on any tools that implement the [Dev container specification](../docs/specs/devcontainer-reference.md).
+- Users can validate the origin of the template or feature that they want to use.
+- Users can pin a particular version of a feature to allow consistent, repeatable environments.
+
+# Proposal
+
+## Github Topics
+
+A set of discoverable dev container-related repos can be achieved with the use of [GitHub Topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics). You may explore repos that GitHub has automatically considered part of the [devcontainers](https://github.com/topics/devcontainers) topic. If you'd like your template or feature repo to be easily discoverable to the community, you may add the **devcontainers-templates** and/or **devcontainers-features** topic. You may read more on how to do this in the [topics guide](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics).
+
+Any UI application can leverage this search to provide a list of repositories that contain features or templates and present them to their users. The application would query the latest release of repos within the list and obtain their details to show them to the users.
+
+An implementation of the spec can also use other mechanisms to speed up or improve the user experience while searching.
+
+At the same time, it should include a way for users to directly reference a repository containing the features or templates they want to use.
+
+## Releases
+
+As a distribution and storage mechanism, we propose using GitHub releases. The releases would contain compressed files with the source code of the feature or template. Included with the release the corresponding metadata file would be either **devcontainer-feature.json**, **devcontainer-template.json** or **devcontainer-collection.json**.
```

Comment:
We've decided to have a single `devcontainer-collection.json` file that contains top level attributes for `features` and `templates`.

We should note, this file will be auto-generated from the contents in the repo (from the `devcontainers/action` GitHub action, or other spec-compliant tooling)